interceptors/options.go

```diff
@@ -1,32 +1,83 @@
 package interceptors
 
-import "google.golang.org/grpc"
+import (
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
 
-type clientOption interface {
+type hystrixOption interface {
 	grpc.CallOption
-	process(*clientOptions)
+	process(*hystrixOptions)
 }
 
-type clientOptions struct {
-	hystrixName string
+type hystrixOptions struct {
```

Comment:
i think `clientOptions` is probably more appropriate since the options may not necessarily be hystrix related in the future, but not a blocker since it's a private implementation detail

Reply:
I thought about it, but I would prefer changing it to hystrix-specific for now because I can't think of any use case for other existing interceptors to utilize those options yet. So in the future, if we need generic options for most interceptors, the clientOptions would be a good timing to show up again then.